### PR TITLE
Add missing property and datatypes for SERVERPROPERTY

### PR DESCRIPTION
**Pull request by srutzky**
1. `IsTempdbMetadataMemoryOptimized` property, added in SQL Server 2019, was missing, but is not "undocumented" given that it's documented in the "TempDB Database" documentation:
    https://docs.microsoft.com/en-us/sql/relational-databases/databases/tempdb-database#memory-optimized-tempdb-metadata
   **PLEASE verify relative URL for the "TempDB Database" link in the new row for `IsTempdbMetadataMemoryOptimized`.**

2. Moved the three "Filestream" properties up to be in alphabetical order, like the rest of the items in the list.

3. Added missing "base data type"s for various properties. The query below shows what the datatypes are. For `ProductBuildType`, I could only get that to not be `NULL` using SQL Server 2012 Express Edition LocalDB (SP4-GDR) (KB4057116) - 11.0.7462.6 (X64).

    ```sql
    SELECT tab.[Property],
           SERVERPROPERTY(tab.[Property]) AS [Value],
           SQL_VARIANT_PROPERTY(SERVERPROPERTY(tab.[Property]), 'BaseType') AS [Datatype],
           SQL_VARIANT_PROPERTY(SERVERPROPERTY(tab.[Property]), 'MaxLength') AS [MaxLength]
    FROM   (VALUES
                   ('IsTempDbMetadataMemoryOptimized'),
                   ('HadrManagerStatus'),
                   ('InstanceDefaultDataPath'),
                   ('InstanceDefaultLogPath'),
                   ('IsAdvancedAnalyticsInstalled'),
                   ('IsBigDataCluster'),
                   ('IsLocalDB'),
                   ('ProductBuild'),
                   ('ProductBuildType'),
                   ('ProductMajorVersion'),
                   ('ProductMinorVersion'),
                   ('ProductUpdateLevel'),
                   ('ProductUpdateReference'),
                   ('FilestreamShareName'),
                   ('FilestreamConfiguredLevel'),
                   ('FilestreamEffectiveLevel')
           ) tab([Property])
    ORDER BY tab.[Property];
    ```

Take care,
Solomon...
https://SqlQuantumLift.com/
https://SqlQuantumLeap.com/
https://SQLsharp.com/

### CODE REVIEW

**Comment by PRMerger9** [comment]
@srutzky : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by VanMSFT** [comment]
@srutzky - Hey Solomon! Thanks for these additions. Let me check on them before approval.

File: docs/t-sql/functions/serverproperty-transact-sql.md [hunk 1]
```diff
@@ -58,32 +58,36 @@ Is an expression that contains the property information to be returned for the s
...
 |IsPolyBaseInstalled|**Applies to**: [!INCLUDE[ssCurrent](../../includes/sscurrent-md.md)].<br /><br /> Returns whether the server instance has the PolyBase feature installed.<br /><br /> 0 = PolyBase is not installed.<br /><br /> 1 = PolyBase is installed.<br /><br /> Base data type: **int**|  
 |IsSingleUser|Server is in single-user mode.<br /><br /> 1 = Single user.<br /><br /> 0 = Not single user<br /><br /> NULL = Input is not valid, an error, or not applicable.<br /><br /> Base data type: **int**|  
+|IsTempDbMetadataMemoryOptimized|**Applies to**: [!INCLUDE[ssSQL2019](../../includes/sssqlv15-md.md)] and later.<br /><br />Returns 1 if TempDB has been enabled to use memory-optimized tables for metadata; 0 if TempDB is using regular, disk-based tables for metadata. For more information, see [TempDB Database](../../relational-databases/databases/tempdb-database#memory-optimized-tempdb-metadata).<br /><br /> Base data type: **int**|
```
Comment (bluefooted) [inline docs/t-sql/functions/serverproperty-transact-sql.md:77]: This looks fine, just request a change to the case of tempdb. We are standardizing on tempdb vs. TempDB, I've already made the changes to the tempdb Database article linked in your submission.

Comment (VanMSFT) [reply docs/t-sql/functions/serverproperty-transact-sql.md]: Thanks Pam! I've updated this PR.

**Comment by VanMSFT** [comment]
@srutzky - Solomon, we're good to go. Thanks again for helping us with these docs!

#sign-off

**Comment by srutzky** [comment]
@VanMSFT (and @bluefooted ): you're welcome 😺 .

**Pull request merged**